### ISSUE TITLE
Readme / cm256cc comment

### ISSUE THREAD

**Issue by dgorissen** [f4exb/sdrdaemon#2]
First of all thanks for this project, looks like exactly what I need. Just a small comment. The readme says:

`To enable the version with FEC (sdrdaemonfec binary and libsdrdmnfec) you have to install CM256cc.`

This led me to assume FEC is optional and needs to be explicitly enabled. However, compilation of a fresh clone complains it can't find CM256 which confused me initially. Compiling/installing and setting the paths with ccmake then fixes it. I guess its trying to build the FEC binaries by default. Otherwise sdrdaemon seems to compile and stream fine. Now I just have to figure out the data format to get a python client working.

**Comment by rajibeee** [comment]
@dgorissen Can you please tell me where did you install cm256cc and set path ( how to set path ? ) 

I am having trouble installing sdrdaemon. Although I have installed cm256cc in the root ( by git clone ) , while doing cmake in sdrdaemon/build the cmake can not find cm256cc :-) What should I do ? More information here ..   [https://github.com/f4exb/sdrdaemon/issues/5](url)

**Comment by dgorissen** [comment]
@rajibeee I just followed the instructions exactly as listed in the readme of sdrdaemon and cm256cc, nothing special was done or needed my end.

**Comment by rajibeee** [comment]
@dgorissen Do you have any idea then why I am getting all these errors ? I am even using a fresh OS on my BeagleBoneBlack.

```
root@beaglebone:~# ls
Desktop  cm256cc  sdrdaemon
root@beaglebone:~# cd sdrdaemon/
root@beaglebone:~/sdrdaemon# cd build/
root@beaglebone:~/sdrdaemon/build# cmake ..
-- Architecture: armv7l
-- Use g++ NEON SIMD instructions
-- Boost version: 1.55.0
-- checking for module 'libcm256cc'
--   package 'libcm256cc' not found
-- CM256cc not found
-- Could NOT find CM256CC (missing:  CM256CC_LIBRARIES CM256CC_INCLUDE_DIR)
-- checking for module 'libairspy'
--   package 'libairspy' not found
-- libairspy not found
-- checking for module 'libbladerf'
--   package 'libbladerf' not found
-- libbladerf not found
-- checking for module 'libhackrf'
--   package 'libhackrf' not found
-- libhackrf not found
-- Found librtlsdr: /usr/include, /usr/lib/arm-linux-gnueabihf/librtlsdr.so
-- Found libusb: /usr/include/libusb-1.0, /usr/lib/arm-linux-gnueabihf/libusb-1.0.so
-- Found libnanomsg: /usr/include, /usr/lib/arm-linux-gnueabihf/libnanomsg.so
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CM256CC_INCLUDE_DIR (ADVANCED)
   used as include directory in directory /root/sdrdaemon
   used as include directory in directory /root/sdrdaemon
   used as include directory in directory /root/sdrdaemon
   used as include directory in directory /root/sdrdaemon
   used as include directory in directory /root/sdrdaemon

-- Configuring incomplete, errors occurred!
See also "/root/sdrdaemon/build/CMakeFiles/CMakeOutput.log".
See also "/root/sdrdaemon/build/CMakeFiles/CMakeError.log".
```

**Comment by dgorissen** [comment]
Just follow the instructions for setting `CM256CC_INCLUDE_DIR` per the readme (section `Forward Erasure Correction (FEC) support`)

**Comment by rajibeee** [comment]
@dgorissen Thanks for your quick replies. Really appreciate it. I have managed to go past `cmake` this time. But at `Make` I am having the following error. Can you please see what's wrong. 

```
login as: root
Debian GNU/Linux 8
Last login: Fri May 16 11:19:05 2014 from 192.168.7.1
root@beaglebone:~# ls
Desktop  cm256cc  sdrdaemon
root@beaglebone:~# cd sdrdaemon/
root@beaglebone:~/sdrdaemon# ls
CMakeLists.txt  build         gr-sdrdaemonfec  sdrdaemon.cpp
README.md       cmake         include          sdrdaemonfec.cpp
TODO.txt        gr-sdrdaemon  sdmnbase         sdrdmnctl.cpp
root@beaglebone:~/sdrdaemon# cd build/
root@beaglebone:~/sdrdaemon/build# make
[ 37%] Built target sdmnbase
[ 79%] Built target sdmnfecbase
[ 83%] Building CXX object CMakeFiles/sdmnrtlsdr.dir/sdmnbase/RtlSdrSource.cpp
Linking CXX static library libsdmnrtlsdr.a
[ 83%] Built target sdmnrtlsdr
[ 87%] Building CXX object CMakeFiles/sdmntest.dir/sdmnbase/TestSource.cpp.o
Linking CXX static library libsdmntest.a
[ 87%] Built target sdmntest
Scanning dependencies of target sdrdaemon
[ 91%] Building CXX object CMakeFiles/sdrdaemon.dir/sdrdaemon.cpp.o
Linking CXX executable sdrdaemon
[ 91%] Built target sdrdaemon
Scanning dependencies of target sdrdaemonfec
[ 95%] Building CXX object CMakeFiles/sdrdaemonfec.dir/sdrdaemonfec.cpp.o
make[2]: *** No rule to make target '/opt/install/cm256cc/lib/libcm256cc.so', needed by 'sdrdaemonfec'.  Stop.
CMakeFiles/Makefile2:239: recipe for target 'CMakeFiles/sdrdaemonfec.dir/all' failed
make[1]: *** [CMakeFiles/sdrdaemonfec.dir/all] Error 2
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2


```
Am I doing anything wrong ?

**Comment by dgorissen** [comment]
Are you sure you compiled / installed libcm256cc correctly and the so file is where cmake/make thinks it is?

**Comment by rajibeee** [comment]
@dgorissen Thanks you again for your reply. I found out that I did not `make` the cm256cc as it was not mentioned in the instructions :-( sorry for being a total newbie. You have been really helpful. Thanks a lot !!

**Comment by f4exb** [comment]
The CMakeLists.txt file has been fixed in v2.1.3 so that it does not fail when cm256cc is not available
